File: Java/ex14.java
class Configuracao {
    private static Configuracao instancia;
    private String configuracao;

    private Configuracao() {
        configuracao = "Configuração padrão";
    }

    public static Configuracao getInstancia() {
        if (instancia == null) {
            instancia = new Configuracao();
        }
        return instancia;
    }

    public void setConfiguracao(String configuracao) {
        this.configuracao = configuracao;
    }

    public String getConfiguracao() {
        return configuracao;
    }
}

public class Main {
    public static void main(String[] args) {
        Configuracao config1 = Configuracao.getInstancia();
        config1.setConfiguracao("Host: localhost");

        Configuracao config2 = Configuracao.getInstancia();
        config2.setConfiguracao("Port: 8080");

        System.out.println(config1.getConfiguracao());
        System.out.println(config2.getConfiguracao());
        System.out.println(config1 == config2);
    }
}

File: Golang/ex14.go
package main

import (
	"fmt"
	"sync"
)

type Configuracao struct {
	configuracao string
}

var (
	instancia *Configuracao
	once      sync.Once
)

func GetConfiguracao() *Configuracao {
	once.Do(func() {
		instancia = &Configuracao{configuracao: "Configuração padrão"}
	})
	return instancia
}

func (c *Configuracao) SetConfiguracao(configuracao string) {
	c.configuracao = configuracao
}

func (c *Configuracao) GetConfiguracao() string {
	return c.configuracao
}

func main() {
	config1 := GetConfiguracao()
	config1.SetConfiguracao("Host: localhost")

	config2 := GetConfiguracao()
	config2.SetConfiguracao("Port: 8080")

	fmt.Println(config1.GetConfiguracao())
	fmt.Println(config2.GetConfiguracao())
	fmt.Println(config1 == config2)
}
